### PR TITLE
feat: executive profile for CEO writing voice

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- Adds `ExecutiveProfileService` for defining the CEO's writing voice (tone, formality, patterns, vocabulary, sign-off), separate from the assistant's persona (office identity)
- New `config/executive-profile.yaml` + DB versioning + hot reload + audit trail — same lifecycle as office identity
- Injected into coordinator system prompt as `${executive_voice_block}` on every turn, replacing the vague "write in the CEO's voice" instruction with structured guidance
- HTTP API at `/api/executive` (GET/PUT/history/reload) for programmatic access
- Fixes pre-existing bug where autonomy block injection in runtime.ts silently discarded identity block substitutions

## Design decisions

- **Executive identity (name, title, org) lives in the contact system** — the profile is purely about style preferences, avoiding redundant storage
- **Non-fatal initialization** — if the config is missing, drafts use generic voice rather than crashing
- **Separate from office identity** — "how the assistant presents itself" vs "how the system represents the executive" are fundamentally different concerns
- **Free-form tone descriptors** — no predefined set; the executive's voice is personal and shouldn't be artificially constrained
- **Opt-in for specialist agents** — v1 only injects into coordinator; any agent can include `${executive_voice_block}` in its system_prompt

## Test plan

- [x] Build passes (`npm run build`)
- [x] All 1488 unit tests pass, 0 regressions
- [x] New tests: YAML schema validation, prompt compilation output format, interpolateRuntimeContext placeholder replacement, coordinator.yaml placeholder presence
- [ ] Boot with `config/executive-profile.yaml` present — verify service initializes, DB seeded
- [ ] Boot without the file — verify warning logged, coordinator functions normally
- [ ] Edit YAML while running — verify hot reload triggers
- [ ] Draft a reply request — verify voice guidance appears in coordinator prompt
- [ ] PUT `/api/executive` — verify new version saved, audit event emitted

## Release Notes

**New Features**
* Executive profile management feature enabling definition and management of CEO writing voice preferences
* New API endpoints for retrieving, updating, and viewing executive profile history
* Two new conversational skills for reading and updating executive profile without manual YAML editing
* Enhanced coordinator system prompts now automatically inject executive voice guidance into all turns

### CODE REVIEW

**Comment by josephfung** [comment]
@Coderabbitai review please

File: skills/executive-profile-update/handler.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+export class ExecutiveProfileUpdateHandler implements SkillHandler {
+  async execute(ctx: SkillContext): Promise<SkillResult> {
+    if (!ctx.executiveProfileService) {
+      return { success: false, error: 'executive-profile-update requires executiveProfileService in context. Is the executive profile configured?' };
+    }
+
+    const { writing_voice } = ctx.input as { writing_voice?: PartialWritingVoiceInput };
+    if (!writing_voice || typeof writing_voice !== 'object') {
+      return { success: false, error: 'Input must include a "writing_voice" object with the fields to update.' };
+    }
```
[the remark this answers is not in the document]
Comment (josephfung) [reply skills/executive-profile-update/handler.ts:33]: Fixed in 7940827. Added `Array.isArray(writing_voice)` check — good catch, `typeof [] === 'object'` is a classic footgun.

File: skills/executive-profile-update/handler.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      // Build a human-readable changes summary before updating.
+      const changes = buildChangesSummary(currentVoice, merged);
+
+      // Persist — validation happens inside update().
+      const changedBy = ctx.caller?.contactId ?? ctx.caller?.role ?? 'system';
+      await ctx.executiveProfileService.update(
+        { writingVoice: merged },
+        changedBy,
+        `Skill update: ${changes}`,
+      );
```
[the remark this answers is not in the document]
Comment (josephfung) [reply skills/executive-profile-update/handler.ts]: Fixed in 7940827. Changed to a fixed `'skill'` source label with the actor identity in the note field. Keeps the audit trail filterable while preserving attribution.

File: src/channels/http/routes/executive.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      return reply.status(400).send({ error: 'Request body must include a "profile" object' });
+    }
+
+    try {
+      const VALID_CHANGED_BY = ['api', 'wizard'] as const;
+      type ChangedBy = typeof VALID_CHANGED_BY[number];
+      const changedBy: ChangedBy = VALID_CHANGED_BY.includes(body.changedBy as ChangedBy)
+        ? (body.changedBy as ChangedBy)
+        : 'api';
+      await executiveProfileService.update(body.profile, changedBy, body.note);
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/channels/http/routes/executive.ts:68]: This matches the identity routes pattern exactly — the wizard UI sends `changedBy: 'wizard'` and the API defaults to `'api'`. The whitelist only accepts those two values, so callers cannot inject arbitrary labels. Hardcoding to `'api'` would break the wizard flow when it is extended to cover executive profile setup. Leaving as-is for parity with the identity routes.

File: src/executive/service.ts [hunk 4]
```diff
@@ -0,0 +1,493 @@
...
+    // Try to load from DB first — DB takes precedence over the file.
+    const dbProfile = await this.loadFromDb();
+    if (dbProfile) {
+      this.cached = dbProfile;
+      this.logger.info('Executive profile loaded from DB');
+    } else {
+      // No DB record — seed from YAML file as version 1.
+      const fileProfile = this.loadFromFile();
+      await this.update(fileProfile, 'file_load', 'Initial load from config/executive-profile.yaml');
+      this.logger.info('Executive profile seeded from config/executive-profile.yaml (version 1)');
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/executive/service.ts:107]: The non-fatal handling is at the bootstrap level in `index.ts`, not the service level — and that's intentional. The service itself is strict (matching `OfficeIdentityService`), and `index.ts` wraps the call in a try/catch that sets `executiveProfileService = undefined` and logs at error level. The process stays alive, drafts use generic voice, and the API routes simply aren't registered.

The concern about "later recovery by adding the file" is valid — the file watcher won't be running since init failed. But the HTTP API (`PUT /api/executive`) and the `executive-profile-update` skill both work without the YAML file, so the profile can be created at runtime through conversation or the wizard. Restart is only needed if you specifically want file-watcher hot reload.

File: src/executive/service.ts [hunk 5]
```diff
@@ -0,0 +1,493 @@
...
+      throw new Error(`writingVoice.formality must be an integer between 0 and 100; got ${voice.formality}`);
+    }
+
+    if (!Array.isArray(voice.patterns)) {
+      throw new Error('writingVoice.patterns must be an array');
+    }
+
+    if (!voice.vocabulary || !Array.isArray(voice.vocabulary.prefer) || !Array.isArray(voice.vocabulary.avoid)) {
+      throw new Error('writingVoice.vocabulary must have prefer and avoid arrays');
+    }
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/executive/service.ts]: Fixed in 7940827. All array fields (tone, patterns, vocabulary.prefer, vocabulary.avoid) now validate that every element is a string.

File: src/index.ts [hunk 6]
```diff
@@ -359,6 +377,27 @@ async function main(): Promise<void> {
...
+         WHERE ci.channel = 'email' AND ci.channel_identifier = $1`,
+        [config.ceoPrimaryEmail],
+      );
+      if (nameResult.rows[0]?.display_name) {
+        executiveDisplayName = nameResult.rows[0].display_name;
+      }
+    } catch (err) {
+      logger.warn({ err }, 'Failed to look up CEO display name for executive voice block — using fallback');
+    }
+  }
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/index.ts:399]: Acknowledged — this is a known trade-off noted in the PR description. Contact renames are rare, and the name is only used in the compiled prompt header ("When drafting under X's name..."). The hot-reloaded profile itself (tone, patterns, vocabulary) is always fresh. A future improvement could query the contact on each turn or subscribe to contact update events, but the complexity doesn't justify itself for a rename-frequency edge case right now.

File: tests/unit/executive/service.test.ts [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+  const dir = fs.mkdtempSync(path.join(os.tmpdir(), 'curia-exec-test-'));
+  const filePath = path.join(dir, 'executive-profile.yaml');
+  fs.writeFileSync(filePath, yaml.dump(content));
+  return { dir, path: filePath };
+}
+
+// Helper: clean up temp directory.
+function cleanup(dir: string): void {
+  fs.rmSync(dir, { recursive: true });
+}
```
[the remark this answers is not in the document]
Comment (josephfung) [reply tests/unit/executive/service.test.ts]: Fixed in 7940827. Removed `writeTempYaml`, `cleanup`, and the `node:os` import.

File: tests/unit/executive/service.test.ts [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+        tone: ['direct', 'warm', 'confident'],
+        formality: 50,
+        patterns: [],
+        vocabulary: { prefer: [], avoid: [] },
+        signOff: '',
+      },
+    };
+    expect(profile.writingVoice.tone).toHaveLength(3);
+  });
+});
```
[the remark this answers is not in the document]
Comment (josephfung) [reply tests/unit/executive/service.test.ts:81]: Fair criticism. The validation tests assert properties of constructed data rather than exercising `validateProfile()`, and the compiler tests duplicate the production logic. The root cause is that both methods are private and the service requires a DB pool to construct.

The skill handler tests (`executive-profile-get.test.ts` and `executive-profile-update.test.ts`) do exercise the real production code through mock service injection, so the key behaviors are covered. The service-level tests are supplementary coverage for the YAML schema shape and compiled output format.

Extracting `compileWritingVoiceBlock` and `validateProfile` as importable pure functions (or a static method) would be the right fix — leaving that for a follow-up since it's a refactor that doesn't change behavior.

[the remark this answers is not in the document]
Comment (josephfung) [reply tests/unit/executive/service.test.ts:81]: @coderabbit yes please, thanks
